src/Core/CFamily/EffectiveRulesConfigCalculator.cs

```diff
@@ -64,32 +64,34 @@ public EffectiveRulesConfigCalculator(ILogger logger)
             configCache = new RulesConfigCache();
         }
 
-        public ICFamilyRulesConfig GetEffectiveRulesConfig(string languageKey, ICFamilyRulesConfig defaultRulesConfig, RulesSettings customRulesConfiguration)
+        public ICFamilyRulesConfig GetEffectiveRulesConfig(string languageKey, ICFamilyRulesConfig defaultRulesConfig, RulesSettings customSettings)
         {
             if (defaultRulesConfig == null)
             {
                 throw new ArgumentNullException(nameof(defaultRulesConfig));
             }
-
-            // Optimisation - if there are no custom rules settings then just return the default
-            if ((customRulesConfiguration?.Rules?.Count ?? 0) == 0)
+            var effectiveConfig = configCache.FindConfig(languageKey, defaultRulesConfig, customSettings);
+            if (effectiveConfig != null)
             {
-                logger.WriteLine(CoreStrings.EffectiveRules_NoCustomRulesSettings);
-                return defaultRulesConfig;
+                logger.WriteLine(CoreStrings.EffectiveRules_CacheHit);
+                return effectiveConfig;
             }
 
-            var effectiveConfig = configCache.FindConfig(languageKey, defaultRulesConfig, customRulesConfiguration);
-            if (effectiveConfig != null)
+            if ((customSettings?.Rules?.Count ?? 0) == 0)
```

Comment:
I think this IF is redundant now.
We will always add to cache now, so I think it makes more sense to have this check inside DynamicCFamilyRulesConfig, or in the caller that passes customSettings.

Reply:
If we log it in the `CFamilyRulesConfigProvider` it will be logged for every analysis not just when a  new cache entry is being created, which is noisy. The `DynamicCFamilyRulesConfig` doesn't currently log anything, so moving the logging there would mean further changes that are pretty low value. 

This class doesn't know where it's being called from it needs to do some sort of check: an assert, an exception, or defensive coding. The same applies to `DynamicCFamilyRulesConfig`.